libnetwork/etchosts/hosts.go

```diff
@@ -243,7 +244,14 @@ func parseExtraHosts(extraHosts []string) (HostEntries, error) {
 		if values[1] == "" {
 			return nil, fmt.Errorf("IP address in host entry %q is empty", entry)
 		}
-		e := HostEntry{IP: values[1], Names: []string{values[0]}}
+		if values[1] == HostGateway && hostContainersInternalIP == "" {
+			return nil, fmt.Errorf("unable to replace %q of host entry %q: host containers internal IP address is empty", HostGateway, entry)
```

Comment:
sorry I just took another look, can you move this into the `if values[1] == HostGateway {` branch below. There is no reason to check the same condition twice.